src/thor/map_matcher.cc

```diff
@@ -129,30 +129,29 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
                      const std::shared_ptr<sif::DynamicCost>* mode_costing,
                      const sif::TravelMode mode,
                      std::vector<std::pair<GraphId, GraphId>>& disconnected_edges,
-                     const DirectionsOptions options) {
+                     const Options options) {
   // Set costing based on the mode
   const auto& costing = mode_costing[static_cast<uint32_t>(mode)];
-  const bool use_timestamps = options.use_timestamps();
+  bool use_timestamps = options.use_timestamps();
 
   // Iterate through the matched path. Form PathInfo - populate elapsed time
   // Return an empty path (or throw exception) if path is not connected.
-  float elapsed_time = 0;
+  Cost elapsed;
```

Comment:
I changed the signature of `PathInfo` to take both the elapsed time and also elapsed cost. because of this there were numerous places where we form a path were we also need a cost. So i reused the `Cost` class since it does this and has some nice arithmetic operators. i also refactored the timestamp use stuff to keep track of the cost and time and then override the time with the user-supplied time when requested.